Feature/111 registration flow

CalderBriar opened the pull request:
1.I temporarly fixed the bug 215 - "On ‘go back’ action in Trip Overview, loading forever" by adding a StartupScreen
2.Register and Login were split
3.Register has new layout with added confirmPassword Input
4.Fixed some logic errs
5.Added drawer as an additional panel (available from "My trips" screen by swiping from the left corner of the screen to the right)
6. Added custom options to drawer navigation to enable adding custom items to the drawer's list

weronikaolejniczak commented on src/domains/user/containers/Register.js:
```diff
@@ -76,7 +76,7 @@ const Auth = (props) => {
     setError(null);
     setIsLoading(true);
-/** 
+    /**
```
Please delete commented code.

weronikaolejniczak commented on src/domains/user/containers/Auth.js:
```diff
@@ -156,7 +156,8 @@ const Auth = (props) => {
             <TouchableOpacity
               onPress={() => {
-                setIsSignup((prevState) => !prevState);
+                //setIsSignup((prevState) => !prevState);
```
Comment has to be deleted.

weronikaolejniczak commented on src/routes.js:
```diff
@@ -32,9 +32,19 @@ import Map from 'map/containers/Map';
 import Weather from 'weather/containers/Weather';
 
 import Colors from 'constants/Colors';
+//import DrawerNavigator from './DrawerNavigator';
```
Comment can be deleted.

weronikaolejniczak commented on src/routes.js:
```diff
@@ -104,7 +114,13 @@ export default function Navigation() {
   );
 }
-
+/** 
```
Same here - delete comment.

weronikaolejniczak commented on src/domains/user/containers/Auth.js:
```diff
@@ -66,6 +66,12 @@ const Auth = (props) => {
       Alert.alert('An error occured!', error, [{text: 'Okay'}]);
     }
   }, [error]);
+/**
```
Delete comment.

weronikaolejniczak commented on src/state/user/userActions.js:
```diff
@@ -1,6 +1,6 @@
 import {AsyncStorage} from 'react-native';
-
+//import auth from '@react-native-firebase/auth'
 //export const SIGNUP = 'SIGNUP';
```
If not used, delete these constants.

mkowalczyk97 commented:
Looks good,

drawer need more styling obv.

weronikaolejniczak reviewed (changes requested):
Couple of comments to delete. Make sure there are no console.log() statements.

mkowalczyk97 commented:
Looks good, delete not used imports and console.logs

drawer need more styling obv.

weronikaolejniczak commented:
Try to put the correct ticket ID for the branch and mention it in the comment. Each subtask of a story should be a separate branch.

CalderBriar commented:
I implemented the requested changes and posted the correct ticketID (272) in the opening merge comment.

Additionally I added less specific error message to handle failed login attempts as was suggested.
